plugin/tasks.py

```diff
@@ -36,76 +36,98 @@ def _decorator(*args, **kwargs):
 @operation
 @throw_cloudify_exceptions
 def create_instance(config, **kwargs):
+    ctx.logger.info('Create instance')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
                               config['scope'],
                               ctx.logger)
 
-    response = gcp.create_instance(ctx.node.name,
-                                   network=config['network'],
+    hostname = utils.get_gcp_resource_name(ctx.instance.id)
+    network = utils.get_gcp_resource_name(config['network'])
+    response = gcp.create_instance(hostname,
+                                   network=network,
                                    agent_image=config['agent_image'])
     gcp.wait_for_operation(response['name'])
+    ctx.instance.runtime_properties['hostname'] = hostname
     set_ip(gcp)
 
 
 @operation
 @throw_cloudify_exceptions
 def delete_instance(config, **kwargs):
+    ctx.logger.info('Delete instance')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
                               config['scope'],
                               ctx.logger)
-    response = gcp.delete_instance(ctx.node.name)
+    response = gcp.delete_instance(ctx.instance.runtime_properties['hostname'])
     gcp.wait_for_operation(response['name'])
 
 
 @operation
 @throw_cloudify_exceptions
 def create_network(config, **kwargs):
+    ctx.logger.info('Create network')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
                               config['scope'],
                               ctx.logger)
-    response = gcp.create_network(config['network'])
+    network_name = utils.get_gcp_resource_name(config['network'])
+    response = gcp.create_network(network_name)
     gcp.wait_for_operation(response['name'], True)
 
 
 @operation
 @throw_cloudify_exceptions
 def delete_network(config, **kwargs):
+    ctx.logger.info('Delete network')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
                               config['scope'],
                               ctx.logger)
-    response = gcp.delete_network(config['network'])
+    network_name = utils.get_gcp_resource_name(config['network'])
+    response = gcp.delete_network(network_name)
     gcp.wait_for_operation(response['name'], True)
```

Comment:
here, instead of recalculating network_name, read the value set to the runtime property during create.
after the network is deleted, remove this runtime property.